nav2_smac_planner/src/smac_planner_2d.cpp

```diff
@@ -285,6 +293,100 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
   return plan;
 }
 
+void SmacPlanner2D::on_parameter_event_callback(
```

Comment:
There's a slightly cleaner way you can handle this that was added in Galactic and newer
- https://github.com/ros2/demos/pull/486/files
- https://docs.ros.org/en/rolling/Tutorials/Monitoring-For-Parameter-Changes-CPP.html

Using the `ParameterEventHandler`, you can add callbacks for specific parameters rather than 1 big callback to rule them all. While I think this is probably the right direction, I would be open if you thought differently before asking you to do that instead. But if you agree that would be cleaner / nicer to work with, that's what I would suggest doing.

Reply:
It looks like you might have to store the callback handle (?) which could make this annoying to deal with. I'm uncertain if you have to store that if it that would be nice in case you needed it.

Reply:
Hum, it doesn't seem ideal either: need to store one handle per custom callback plus in our case we would have to define 3 custom callback: 1. for tolerance, 2. for the a_star group of params, and 3. for the downsampler group of param.
I would leave it like that. I am not a super fan of the current approach but the new one doesn't satisfy me more

Reply:
OK sounds good